core/src/HIP/Kokkos_HIP_Parallel_MDRange.hpp

```diff
@@ -296,6 +299,44 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
     }
   }
 
+  __device__ inline void run(ShflReductionTag) const {
+    value_type value;
+    ValueInit::init(ReducerConditional::select(m_functor, m_reducer), &value);
+    // Number of blocks is bounded so that the reduction can be limited to two
+    // passes. Each thread block is given an approximately equal amount of work
+    // to  perform. Accumulate the values for this block. The accumulation
+    // ordering does not match the final pass, but is arithmetically equivalent.
+
+    const Member work_part =
+        ((m_policy.m_num_tiles + (gridDim.x - 1)) /
+         gridDim.x);  // portion of tiles handled by each block
+
+    this->exec_range(value);
+
+    pointer_type const result = m_scratch_space;
+
+    int max_active_thread = work_part < blockDim.y ? work_part : blockDim.y;
+    max_active_thread =
+        (max_active_thread == 0) ? blockDim.y : max_active_thread;
+
+    value_type init;
+    ValueInit::init(ReducerConditional::select(m_functor, m_reducer), &init);
+    if (Impl::hip_inter_block_reduction<ReducerTypeFwd, ValueJoin, WorkTagFwd>(
+            value, init,
+            ValueJoin(ReducerConditional::select(m_functor, m_reducer)),
+            m_scratch_space, result, m_scratch_flags, max_active_thread)) {
+      const unsigned id = threadIdx.y * blockDim.x + threadIdx.x;
```

Comment:
```suggestion
      unsigned const id = threadIdx.y * blockDim.x + threadIdx.x;
```

Reply:
No that is not required. I like const unsigned much more :-)